pkg/operator/controller/credentials-secret/controller.go

```diff
@@ -52,24 +58,30 @@ type Config struct {
 }
 
 type reconciler struct {
-	cache  cache.Cache
-	scheme *runtime.Scheme
-	client client.Client
-	config Config
-	log    logr.Logger
+	cache      cache.Cache
+	scheme     *runtime.Scheme
+	client     client.Client
+	config     Config
+	log        logr.Logger
+	kubeclient discovery.DiscoveryInterface
```

Comment:
The name doesn't seem to reflect well the functionality behind. Something like `kubeDiscovery` would be more relevant, wdyt?